src/plugins/mer/merconstants.h

```diff
@@ -29,11 +29,19 @@
 namespace Mer {
 namespace Constants {
 
+#ifdef MER_BASE_OS_NAME
+const char MER_OS_NAME[] = QT_TRANSLATE_NOOP("Mer", MER_BASE_OS_NAME " OS");
+const char MER_SDK_NAME[] = QT_TRANSLATE_NOOP("Mer", MER_BASE_OS_NAME " SDK");
+#else
+const char MER_OS_NAME[] = "Sailfish OS";
+const char MER_SDK_NAME[] = "Sailfish SDK";
+#endif
+
```

Comment:
This seems wrong - one branch is i18n-enabled, the other is not. Is is necessary to enable i18n for these at all?

This also makes me think, do we really need to add a custom mechanism to change the branding? Wouldn't it be more convenient to use i18n for that. I mean you can leave Sailfish OS/Sailfish SDK in the code and provide translations (easy to automate that) that say something else.

Reply:
martyone, are you saying that there never will be Russian translations for the Sailfish SDK, or english version of the Aurora OS SDK?

Reply:
I was thinking if they need to switch that at run time. I would be surprised if Sailfish OS/SDK was subject to l10n but yeah, it's different with Aurora where their native writing system is likely to be used primarily.

The rest of my questions still applies.

Reply:
> Other way, we can use l10n for all language, where during the build we will change the OS name, but it's harder to maintain

That's what I meant. Except that I do not think it is harder to maintain. Even though I admit it is not the cleanest practice, it should be at least equally reliable and much cheaper at the same time, compared to the obvious solution with string substitution at C++ side. Not mandatory, but worth to consider.

> And i have some other question, can we add to this repo branding icons and translation?

AFAIK this is not allowed.

Reply:
I try to use l10n for it. It's work but we need some change in upstream code to add loading translation for `<System Language>` and `English`. 

But we would prefer to use local variable in c++ code. It is simple to use and maintain with CI

Reply:
OK, let's do it on C++ side.

The definition needs to be done under src/libs/sfdk instead of src/plugins/mer.

I would also prefer exposing functions that return the translated QString instead, so that we do not need to repeat tr(context, ...) everywhere. It could be exposed on Sfdk::Sdk class:

```
     Sdk(Options options = NoOption);
     ~Sdk() override;
     static Sdk *instance();

+    static QString osVariant();
+    static QString sdkVariant();
+
     static void enableUpdates();
     static bool isApplyingUpdates();
     static bool saveSettings(QStringList *errorStrings);
```

The actual strings should come from a header file similarly to how it is done with https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/app/app_version.h.in. Our header file would be private though. `sfdk_version_p.h.in` would work for me as a name for the header.

Reply:
Ok, move it to src/libs/sfdk